mtp_api/apps/mtp_auth/tests/test_views.py

```diff
@@ -146,3 +149,82 @@ def test_bank_admin_cannot_login_to_cashbook(self):
             }
         )
         self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class AccountLockoutTestCase(APITestCase):
+    fixtures = ['test_prisons.json', 'initial_groups.json']
+
+    def setUp(self):
+        super().setUp()
+        self.prison_clerks = make_test_users()[0]
+
+    def pass_login(self, user, client):
+        response = self.client.post(
+            reverse('oauth2_provider:token'),
+            {
+                'grant_type': 'password',
+                'username': user.username,
+                'password': user.username,
+                'client_id': client.client_id,
+                'client_secret': client.client_secret,
+            }
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def fail_login(self, user, client):
+        response = self.client.post(
+            reverse('oauth2_provider:token'),
+            {
+                'grant_type': 'password',
+                'username': user.username,
+                'password': 'incorrect-password',
+                'client_id': client.client_id,
+                'client_secret': client.client_secret,
+            }
+        )
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_account_lockout_on_too_many_attempts(self):
+        prison_clerk = self.prison_clerks[0]
+        cashbook_client = Application.objects.get(client_id=CASHBOOK_OAUTH_CLIENT_ID)
+
+        for _ in range(settings.MTP_AUTH_LOCKOUT_COUNT):
+            self.assertFalse(FailedLoginAttempt.is_locked_out(prison_clerk, cashbook_client))
+            self.fail_login(prison_clerk, cashbook_client)
+
+        self.assertTrue(FailedLoginAttempt.is_locked_out(prison_clerk, cashbook_client))
+        self.fail_login(prison_clerk, cashbook_client)
+
+    def test_account_lockout_removed_on_successful_login(self):
+        if not settings.MTP_AUTH_LOCKOUT_COUNT:
+            return
+
+        prison_clerk = self.prison_clerks[0]
+        cashbook_client = Application.objects.get(client_id=CASHBOOK_OAUTH_CLIENT_ID)
+
+        for _ in range(settings.MTP_AUTH_LOCKOUT_COUNT - 1):
+            self.assertFalse(FailedLoginAttempt.is_locked_out(prison_clerk, cashbook_client))
+            self.fail_login(prison_clerk, cashbook_client)
+
+        self.assertFalse(FailedLoginAttempt.is_locked_out(prison_clerk, cashbook_client))
+        self.pass_login(prison_clerk, cashbook_client)
+        self.assertFalse(FailedLoginAttempt.is_locked_out(prison_clerk, cashbook_client))
+        self.assertEqual(FailedLoginAttempt.objects.filter(
+            user=prison_clerk,
+            application=cashbook_client,
+        ).count(), 0)
+
+    def test_per_application_account_lockout(self):
```

Comment:
This might be clearer about the intent of the test as `test_lockout_only_applies_to_current_application` or something along those lines?